File: src/main/java/com/kmatrokhin/uvbot/services/LocationInfoService.java
package com.kmatrokhin.uvbot.services;

import com.kmatrokhin.uvbot.dto.Coordinates;
import com.kmatrokhin.uvbot.dto.LocationInfo;
import com.kmatrokhin.uvbot.dto.Weather;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class LocationInfoService {
    private final WeatherService weatherService;
    private final GeocodingService geocodingService;

    public LocationInfo getLocationInfo(Coordinates coordinates) {
        Weather weather = weatherService.getWeather(coordinates);
        String locationName = geocodingService.getLocationName(coordinates);
        return new LocationInfo(locationName, coordinates, weather);
    }
}

File: src/main/java/com/kmatrokhin/uvbot/services/LocationInfoService.kt
package com.kmatrokhin.uvbot.services

import com.kmatrokhin.uvbot.dto.Coordinates
import com.kmatrokhin.uvbot.dto.LocationInfo
import com.kmatrokhin.uvbot.entities.UserLanguage
import org.springframework.stereotype.Service

@Service
class LocationInfoService(
    private val weatherService: WeatherService,
    private val geocodingService: GeocodingService
) {
    fun getLocationInfo(
        coordinates: Coordinates,
        language: UserLanguage,
        locationName: String? = null
    ): LocationInfo {
        val weather = weatherService.getWeather(coordinates)
        val resolvedName = locationName ?: geocodingService.getLocationName(coordinates, language)
        return LocationInfo(resolvedName, coordinates, weather)
    }
}
